pkg/util/xorm/session_insert.go

```diff
@@ -345,6 +346,20 @@ func (session *Session) innerInsert(bean any) (int64, error) {
 		return 0, err
 	}
 
+	// XXX: hack to handle autoincrement in spanner
+	if len(table.AutoIncrement) > 0 && session.engine.dialect.DBType() == "spanner" {
+		var found bool
+		for _, col := range colNames {
+			if col == table.AutoIncrement {
+				found = true
+			}
+		}
+		if !found {
+			colNames = append(colNames, table.AutoIncrement)
+			args = append(args, rand.Int63n(9e15))
```

Comment:
Generally speaking using a global rand in a high-throughput place is not a great idea. We might want to have a dedicated per-session random source for this.

Reply:
This is a workaround until we have a better solution (sequence and columns with autogenerated unique IDs).